Mathlib/Topology/Algebra/Category/ProfiniteGrp.lean

```diff
@@ -161,4 +165,169 @@ instance : HasForget₂ ProfiniteGrp Grp where
     map := fun f => f.toMonoidHom
   }
 
+/-- A topological group that has a ContinuousMulEquiv to a profinite group is profinite. -/
+def ofContinuousMulEquivProfiniteGrp {G : ProfiniteGrp.{u}} {H : Type v} [TopologicalSpace H]
+    [Group H] [TopologicalGroup H] (e : ContinuousMulEquiv G H) : ProfiniteGrp.{v} :=
+  letI : CompactSpace H := Homeomorph.compactSpace e.toHomeomorph
+  letI : TotallyDisconnectedSpace G := Profinite.instTotallyDisconnectedSpaceαTopologicalSpaceToTop
+  letI : TotallyDisconnectedSpace H := Homeomorph.totallyDisconnectedSpace e.toHomeomorph
+  .of H
+
+/-- A closed subgroup of a profinite group is profinite. -/
+def ofClosedSubgroup {G : ProfiniteGrp}
+    (H : Subgroup G) (hH : IsClosed (H : Set G)) : ProfiniteGrp :=
+  letI : CompactSpace H := isCompact_iff_compactSpace.mp (IsClosed.isCompact hH)
+  of H
+
+end ProfiniteGrp
+
+/-!
+# The projective limit of finite groups is profinite
+
+* `FiniteGrp.limit` : the concretely constructed limit of finite group as a subgroup of the pi-type
+
+* `ofFiniteGrpLimit`: direct limit of finite groups is a profinite group
+
+* Verify that the constructed limit satisfies the universal property.
+-/
+
+section Profiniteoflimit
+
+open TopologicalGroup
+
+universe w w'
+
+variable {J : Type v} [Category.{w, v} J] (F : J ⥤ FiniteGrp.{max v w'})
+
+attribute [local instance] ConcreteCategory.instFunLike ConcreteCategory.hasCoeToSort
+
+instance (j : J) : TopologicalSpace (F.obj j) := ⊥
+
+instance (j : J) : DiscreteTopology (F.obj j) := ⟨rfl⟩
+
+instance (j : J) : TopologicalGroup (F.obj j) := {}
```

Comment:
This seems a bit strange to me. Why don't you add the local instance that every finite group is a Topological group? BTW: I would put most/all of these material in a namespace, whereas it is now in `_root`.